okio/src/commonMain/kotlin/okio/internal/Buffer.kt

```diff
@@ -1232,3 +1233,47 @@ internal inline fun Buffer.commonHashCode(): Int {
   } while (s !== head)
   return result
 }
+
+/** Returns an immutable copy of this buffer as a byte string.  */
+internal inline fun Buffer.commonSnapshot(): ByteString {
+  check(size <= Int.MAX_VALUE) { "size > Int.MAX_VALUE: $size" }
+  return snapshot(size.toInt())
+}
+
+/** Returns an immutable copy of the first `byteCount` bytes of this buffer as a byte string. */
+internal inline fun Buffer.commonSnapshot(byteCount: Int): ByteString {
+  if (byteCount == 0) return ByteString.EMPTY
+  checkOffsetAndCount(size, 0, byteCount.toLong())
```

Comment:
Moved the implementation of `SegmentedByteString.of` directory into this method to avoid extra expect/actual's.